### PR TITLE
Preliminary work on encryption

### PR DESCRIPTION
**Pull request by jbaublitz**
Supersedes #1705

### CODE REVIEW

**Comment by mulkieran** [comment]
@jbaublitz   Please rebase when you get the chance. It's git's erroneous move of ```create_pool``` when actually it was ```block_evaluate``` that got moved in PR #1767 that is causing the trouble.

**Comment by jbaublitz** [comment]
@mulkieran Rebased.

File: src/dbus_api/api/shared.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        pool::create_dbus_pool,
+        types::TData,
+        util::{
+            engine_to_dbus_err_tuple, get_next_arg, msg_code_ok, msg_string_ok, tuple_to_option,
+        },
+    },
+    engine::CreateAction,
+};
+
+pub fn create_pool_shared(m: &MethodInfo<MTFn<TData>, TData>, has_keyfile: bool) -> MethodResult {
```
[the remark this answers is not in the document]
Comment (jbaublitz) [reply src/dbus_api/api/shared.rs]: Resolved.

**Comment by jbaublitz** [comment]
Conflict resolved.

File: src/engine/strat_engine/backstore/blockdev.rs [hunk 2]
```diff
@@ -60,14 +63,20 @@ impl StratBlockDev {
...
         user_info: Option<String>,
         hardware_info: Option<String>,
+        keyfile_path: Option<&Path>,
     ) -> StratisResult<StratBlockDev> {
         let mut segments = vec![(Sectors(0), bda.extended_size().sectors())];
         segments.extend(upper_segments);
         let allocator = RangeAllocator::new(bda.dev_size(), &segments)?;
 
+        let devnode_maybe_encrypted = match keyfile_path {
+            Some(path) => Self::encrypt_blockdev(devnode.as_path(), path)?,
```
[the remark this answers is not in the document]
Comment (jbaublitz) [reply src/engine/strat_engine/backstore/blockdev.rs]: Resolved.

**Comment by mulkieran** [comment]
Please rebase.

**Comment by jbaublitz** [comment]
Rebased.

**Comment by jbaublitz** [comment]
Closing and creating a new branch off of `develop-2.0.1`.

**Pull request closed without merging**